nixos/modules/system/boot/luksroot.nix

```diff
@@ -212,13 +227,28 @@ let
             ${csopen} --key-file=${dev.keyFile} \
               ${optionalString (dev.keyFileSize != null) "--keyfile-size=${toString dev.keyFileSize}"} \
               ${optionalString (dev.keyFileOffset != null) "--keyfile-offset=${toString dev.keyFileOffset}"}
+            cs_status=$?
+            if [ $cs_status -ne 0 ]; then
+              echo "Key File ${dev.keyFile} failed!"
+              if ! try_empty_passphrase; then
+                ${if dev.fallbackToPassword then "echo" else "die"} "${dev.keyFile} is unavailable"
+                echo " - failing back to interactive password prompt"
+                do_open_passphrase
+              fi
+            fi
```

Comment:
Why are you adding a password check when the key file fails? Before this change, `fallbackToPassword` only did anything if the key file was completely missing. If the file was present but wrong, the device failed.

Reply:
Added this for the case that you would want multiple ways of unlocking a disk